### PR TITLE
John conroy/split bundle vendors

### PR DESCRIPTION
**Pull request by john-conroy**
Splits all shared node modules into own entry chunk named `vendors.bundle.js`. The main bundle has been renamed `main.bundle.js`.

### CODE REVIEW

File: context/build-utils/webpack.common.js [hunk 1]
```diff
@@ -3,11 +3,11 @@ const { resolve } = require('path');
 const { CleanWebpackPlugin } = require('clean-webpack-plugin');
 
 const config = {
-  entry: './app/static/js/index.jsx',
+  entry: { main: './app/static/js/index.jsx' },
   output: {
     path: resolve('./app/static/public'),
-    filename: 'bundle.js',
-    publicPath: resolve('../app/static/public'),
+    publicPath: `${resolve('/static/public/')}/`,
```
Comment (mccalluc) [inline context/build-utils/webpack.common.js:9]: The final "/" probably is required? Would it just be an error if you gave a directory name without the final "/"?

File: context/build-utils/webpack.common.js [hunk 2]
```diff
@@ -86,6 +86,17 @@ const config = {
     ],
   },
   plugins: [new CleanWebpackPlugin()],
+  optimization: {
+    splitChunks: {
+      cacheGroups: {
+        commons: {
+          test: /[\\/]node_modules[\\/]/,
```
Comment (mccalluc) [inline context/build-utils/webpack.common.js:93]:
```suggestion
          test: RegExp('/node_modules/'),
```
- I don't think we need to ensure MS-Windows compatibility?
- Longer syntax is more readable here.

File: context/build-utils/webpack.common.js (hunk 1 above)
Comment (john-conroy) [reply context/build-utils/webpack.common.js:9]: Yeah, not including the additional / was producing an error for whatever reason.

File: context/build-utils/webpack.common.js (hunk 2 above)
Comment (john-conroy) [reply context/build-utils/webpack.common.js:93]: Sure, I just included a suggested snippet.

**Comment by john-conroy** [comment]
@ilan-gold sounds good, I'll do so in the future.

**Pull request merged**